### PR TITLE
feat(base-cluster/tracing): add OpenTelemetry Collector and Grafana Tempo

### PR DESCRIPTION
**Pull request by cwrau**

### CODE REVIEW

File: charts/base-cluster/values.yaml [hunk 1]
```diff
@@ -283,6 +290,8 @@ monitoring:
     enabled: true
   securityScanning:
     enabled: true
+  tracing:
+    enabled: false
```
[the remark this answers is not in the document]
Comment (cwrau) [reply charts/base-cluster/values.yaml:294]: We could enable tracing by default, but I'd guess not many people even have stuff that has tracing, therefore I'd leave this off by default.

File: charts/base-cluster/values.yaml [hunk 2]
```diff
@@ -118,6 +119,7 @@ global:
     kyverno:
       url: https://kyverno.github.io/kyverno
       charts:
+        # TODO: migrate to v3 https://artifacthub.io/packages/helm/kyverno/kyverno#migrating-from-v2-to-v3
         kyverno: 2.x.x
```
[the remark this answers is not in the document]
Comment (cwrau) [reply charts/base-cluster/values.yaml:123]: No, we don't require kyverno v3 for tracing, see https://github.com/teutonet/teutonet-helm-charts/pull/439/files/de30755e09cd0bf309f1885f94bca2a3a73ade11#diff-ac28f53c0d9356ec8584c2df6b227653ffb30381b8d3a75d1738f7e490028037R30

File: charts/base-cluster/templates/kyverno/kyverno.yaml [hunk 3]
```diff
@@ -27,6 +27,22 @@ spec:
       enabled: {{ .Values.monitoring.prometheus.enabled }}
       additionalLabels: {{- .Values.monitoring.labels | toYaml | nindent 8 }}
     priorityClassName: system-cluster-critical
+    # this only works in version 3
+    {{- if and .Values.monitoring.tracing.enabled .Values.monitoring.prometheus.enabled }}
```
[the remark this answers is not in the document]
Comment (cwrau) [reply charts/base-cluster/templates/kyverno/kyverno.yaml:31]: Jein, this only *works* with v3, but this configuration doesn't *need* v3

This still rolls out with v2, it just doesn't work

File: charts/base-cluster/templates/monitoring/tracing/opentelemetry-collector.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  name: open-telemetry-collector
+  namespace: monitoring
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/component: tracing
+    app.kubernetes.io/part-of: monitoring
+spec:
+  chart:
+    spec: {{- include "base-cluster.helm.chartSpec" (dict "repo" "open-telemetry" "chart" "opentelemetry-collector" "context" $) | nindent 6 }}
+  interval: 1h
+  {{- if .Values.monitoring.prometheus.enabled }}
```
Comment (marvinWolff) [inline charts/base-cluster/templates/monitoring/tracing/opentelemetry-collector.yaml:14]: Isn't this check redundant?

In line one the whole Helmrelease is only created when prometheus is enabled, and here you check if prometheus is enabled again to set the `dependsOn`.  But if this statement ever gets evaluatet it's always true, or am i wrong?

File: charts/base-cluster/templates/monitoring/tracing/grafana-tempo.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  name: grafana-tempo
+  namespace: monitoring
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/component: tracing
+    app.kubernetes.io/part-of: monitoring
+spec:
+  chart:
+    spec: {{- include "base-cluster.helm.chartSpec" (dict "repo" "bitnami" "chart" "grafana-tempo" "context" $) | nindent 6 }}
+  interval: 1h
+  {{- if .Values.monitoring.prometheus.enabled }}
```
Comment (marvinWolff) [inline charts/base-cluster/templates/monitoring/tracing/grafana-tempo.yaml:14]: Isn't this check redundant?

In line one the whole Helmrelease is only created when prometheus is enabled, and here you check if prometheus is enabled again to set the dependsOn. But if this statement ever gets evaluatet it's always true, or am i wrong?

**Pull request merged**